src/components/recovery/RecoveryDescription/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { Typography } from '@mui/material'
+import { useMemo } from 'react'
+import type { ReactElement } from 'react'
+
+import EthHashInfo from '@/components/common/EthHashInfo'
+import { InfoDetails } from '@/components/transactions/InfoDetails'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import { useIsGuardian } from '@/hooks/useIsGuardian'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { logError, Errors } from '@/services/exceptions'
+import { getRecoveredSafeInfo } from '@/services/recovery/transaction-list'
+import type { RecoveryQueueItem } from '@/services/recovery/recovery-state'
+
+export function RecoveryDescription({ item }: { item: RecoveryQueueItem }): ReactElement {
+  const { args, isMalicious } = item
+  const { safe } = useSafeInfo()
+  const isGuardian = useIsGuardian()
+
+  const newSetup = useMemo(() => {
+    try {
+      return getRecoveredSafeInfo(safe, {
```

Comment:
This function does not consider some cases
- Swapping an owner, that does not exist anymore
- removing an owner, that does not exist anymore
- adding an owner, that's already added

But: Those calls would all revert anyways.
In general I feel like this is very detailed. Isn't just checking for transaction validity through `isValidExecution` enough?

Reply:
I've added validation for these cases via a static call in 72bcd714ed82c8c2eacf8347d0108135d5a1c9fc.

Reply:
As per discussion: this is an edge case that we won't fix as the current implementation is sufficient for a prediciton combined with the validation checks for execution/cancellation.